src/main/java/xyz/nucleoid/plasmid/world/bubble/BubbleWorldSpawner.java

```diff
@@ -0,0 +1,49 @@
+package xyz.nucleoid.plasmid.world.bubble;
+
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.Heightmap;
+import net.minecraft.world.chunk.Chunk;
+
+public interface BubbleWorldSpawner {
+    static BubbleWorldSpawner at(Vec3d pos) {
+        return new At(pos);
+    }
+
+    static BubbleWorldSpawner atSurface(BlockPos pos) {
```

Comment:
I think an `atSurface(int x, int z)` would be useful in addition to the blockpos version, so we don't need to construct it if we don't have to